app/auth/security.py

```diff
@@ -0,0 +1,77 @@
+from typing import Union, Annotated
+from datetime import timezone, timedelta, datetime
+
+from passlib.context import CryptContext
+import jwt
+from jwt.exceptions import InvalidTokenError
+from fastapi.security import OAuth2PasswordBearer
+from fastapi import HTTPException, Depends, status
+from ..models.users import Users, TokenData
+from app.db import engine
+from sqlmodel import Session
+from app.config import Config
+
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+def verify_password(plain_password, hashed_password):
+    return pwd_context.verify(plain_password, hashed_password)
+
+def get_password_hash(password):
+    return pwd_context.hash(password)
+
+def get_user(username: str):
+    with Session(engine) as session:
+        
+        user = session.get(Users, username)
+        
+        return user
+
+def authenticate_user(username: str, password: str):
+    user = get_user(username)
+    if not user:
+        return False
+    if not verify_password(password, user.hashed_password):
+        return False
+    return user
```

Comment:
I am unsure why this function should return the user object, rather than just returning `True` or `False`.

If it is important could you please add a doc string explaining this, or replace the function with something like this please.

```
def authenticate_user(username: str, password: str):
   """Authenticates the user by checking if the user exists exists in the database 
   and verifying that the password hashes match.
   """
   user = get_user(username)
    if not user:
        return False
    if not verify_password(password, user.hashed_password):
        return False
    return True